### PR TITLE
[Merged by Bors] - feat: Strictly convex functions have unique minima

### PR DESCRIPTION
**Pull request by YaelDillies**

### CODE REVIEW

File: Mathlib/Analysis/Convex/Function.lean [hunk 1]
```diff
@@ -1127,9 +1128,36 @@ theorem OrderIso.concaveOn_symm (f : α ≃o β) (hf : ConvexOn 𝕜 univ f) :
...
+lemma StrictConvexOn.eq_of_isMinOn (hf : StrictConvexOn 𝕜 s f) (hfx : IsMinOn f s x)
+    (hfy : IsMinOn f s y) (hx : x ∈ s) (hy : y ∈ s) : x = y := by
+  by_contra hxy
+  let z := (2 : 𝕜)⁻¹ • x + (2 : 𝕜)⁻¹ • y
+  have hz : z ∈ s := hf.1 hx hy (by norm_num) (by norm_num) $ by norm_num
+  refine lt_irrefl (f z) ?_
+  calc
+    f z < _ := hf.2 hx hy hxy (by norm_num) (by norm_num) $ by norm_num
+    _ ≤ (2 : 𝕜)⁻¹ • f z + (2 : 𝕜)⁻¹ • f z := add_le_add (smul_le_smul_of_nonneg (hfx hz) $ by
+        norm_num) (smul_le_smul_of_nonneg (hfy hz) $ by norm_num)
```
Comment (hrmacbeth) [inline Mathlib/Analysis/Convex/Function.lean:1149]:
```suggestion
    _ ≤ (2 : 𝕜)⁻¹ • f z + (2 : 𝕜)⁻¹ • f z := by gcongr; exacts [hfx hz, hfy hz]
```

**Comment by hrmacbeth** [comment]
bors d+

**Comment by YaelDillies** [comment]
maintainer merge

**Comment by robertylewis** [comment]
bors merge

**Pull request closed without merging**